### PR TITLE
lsvm to display VM details

### PR DESCRIPTION
**Pull request by gurevichmark**
Make `lsvm` command display information / details about a VM. Fixes issue #1023 

Output examples:

1. `lsvm <hypervisor>` -- no changes from current display:

```
[root@fs2vm102 gurevich]# lsvm fs2
fs2: fs2vm101
fs2: fs2vm105
fs2: lsvm_test1
fs2: fs2vm100
fs2: lsvm_test2
fs2: fs2vm104
fs2: fs2vm102
fs2: fs2vm111
[root@fs2vm102 gurevich]#
```

2. `lsvm <vm>` -- new output

```
[root@fs2vm102 gurevich]# lsvm lsvm_test1,lsvm_test2
lsvm_test1: Id: 42
lsvm_test1: Host: fs2
lsvm_test1: OS: hvm
lsvm_test1: Memory: 4194304
lsvm_test1: Virt CPU: 1
lsvm_test1: State: 1 (The domain is active and running)
lsvm_test1: Disk file: /var/lib/libvirt/images/lsvm_test1.sda.qcow2
lsvm_test2: Id: 43
lsvm_test2: Host: fs2
lsvm_test2: OS: hvm
lsvm_test2: Memory: 4194304
lsvm_test2: Virt CPU: 1
lsvm_test2: State: 1 (The domain is active and running)
lsvm_test2: Disk file: /var/lib/libvirt/images/lsvm_test2.sda.qcow2
[root@fs2vm102 gurevich]#
```

### CODE REVIEW

**Comment by whowutwut** [comment]
I will take a review of this next week

File: xCAT-server/lib/xcat/plugins/kvm.pm [hunk 1]
```diff
@@ -3512,11 +3512,73 @@ sub rscan {
...
+                if (exists $domain_info->{"memory"}) {
+                    push @vms, "Memory:" . $domain_info->{"memory"};
+                }
+                if (exists $domain_info->{"nrVirtCpu"}) {
+                    push @vms, "Virt CPU: " . $domain_info->{"nrVirtCpu"};
+                }
+                if (exists $domain_info->{"state"}) {
+                    my $state =  $domain_info->{"state"};
+                    my $state_string = "Unknown";
+                    if ($state == &Sys::Virt::Domain::STATE_NOSTATE) {$state_string = "The domain is active, but is not running / blocked (eg idle)";}
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: Should this be a case statement? and where did we get the mapping of states to txt?

File: xCAT-server/lib/xcat/plugins/kvm.pm [hunk 2]
```diff
@@ -4031,7 +4099,7 @@ sub dohyp {
             return 1, "General error establishing libvirt communication";
         }
     }
-    if (($command eq 'mkvm' or $command eq 'chvm' or $command eq 'rpower') and $hypconn) {
+    if (($command eq 'mkvm' or $command eq 'chvm' or $command eq 'rpower' or $command eq 'lsvm') and $hypconn) {
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: Could we do some test here like in the openbmc.pm code? 
```
unless ($subcommand =~ /^cpu$|^dimm$|^model$|^serial$|^firm$|^mac$|^vpd$|^mprom$|^deviceid$|^guid$|^uuid$|^all$/) {
```

File: xCAT-server/lib/xcat/plugins/kvm.pm [hunk 3]
```diff
@@ -3512,11 +3512,73 @@ sub rscan {
...
+            if ($_->get_name() eq $vm) {
+                push @vms, "Id:" . $_->get_id();
+                push @vms, "Host:" . $host;
+                push @vms, "OS:" . $_->get_os_type();
+                my $domain_info = $_->get_info();
+                if (exists $domain_info->{"memory"}) {
+                    push @vms, "Memory:" . $domain_info->{"memory"};
+                }
+                if (exists $domain_info->{"nrVirtCpu"}) {
+                    push @vms, "Virt CPU: " . $domain_info->{"nrVirtCpu"};
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: Maybe just say "CPUs:" instead of "Virt CPU:"

File: xCAT-server/lib/xcat/plugins/kvm.pm [hunk 4]
```diff
@@ -3512,11 +3512,73 @@ sub rscan {
...
+    my $vm = shift;
     my @doms = $hypconn->list_domains();
     my @vms;
-    foreach (@doms) {
-        push @vms, $_->get_name();
+
+    if ($host ne $vm) {
+        # Processing lsvm for a VM, display details about that VM
+        foreach (@doms) {
+            if ($_->get_name() eq $vm) {
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: How did we come up with the list of attributes to display?

**Comment by whowutwut** [comment]
@immarvin @zet809 If you have no other comments, I will merge this.

File: xCAT-server/lib/xcat/plugins/kvm.pm (hunk 4 above)
Comment (gurevichmark) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: I just looked at the list of available `get_*` functions to see what appeared to be interesting to display

File: xCAT-server/lib/xcat/plugins/kvm.pm (hunk 3 above)
Comment (gurevichmark) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: Changed in the latest commit.

File: xCAT-server/lib/xcat/plugins/kvm.pm (hunk 1 above)
Comment (gurevichmark) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: Replaced `elsif` with `given-when` in the latest commit.
The list of state descriptions came from http://search.cpan.org/dist/Sys-Virt/

File: xCAT-server/lib/xcat/plugins/kvm.pm (hunk 2 above)
Comment (gurevichmark) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: Changed in the latest commit.

File: xCAT-server/lib/xcat/plugins/kvm.pm [hunk 5]
```diff
@@ -4099,7 +4111,8 @@ sub dohyp {
             return 1, "General error establishing libvirt communication";
         }
     }
-    if (($command eq 'mkvm' or $command eq 'chvm' or $command eq 'rpower' or $command eq 'lsvm') and $hypconn) {
+    #if (($command eq 'mkvm' or $command eq 'chvm' or $command eq 'rpower' or $command eq 'lsvm') and $hypconn) {
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: oh, just saw this, we can probably delete this line

Comment (gurevichmark) [inline xCAT-server/lib/xcat/plugins/kvm.pm]: Fixed, getting a little careless. :-(

**Comment by whowutwut** [comment]
Heard no objections, merging in the changes.

**Pull request merged**